analyzers/src/SonarAnalyzer.CSharp/Rules/CryptographicKeyShouldNotBeTooShort.cs

```diff
@@ -98,17 +100,17 @@ protected override void Initialize(SonarAnalysisContext context)
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
-                    var objectCreation = (ObjectCreationExpressionSyntax)c.Node;
-                    if (!(c.SemanticModel.GetTypeInfo(objectCreation).Type is ITypeSymbol containingType))
+                    var objectCreation = ObjectCreationFactory.Create(c.Node);
+                    if (!(c.SemanticModel.GetTypeInfo(objectCreation.Expression).Type is ITypeSymbol containingType))
```

Comment:
Can the condition be covered with nonexistent type?
```
var c = new UndefinedType(); // Error [CS0264]
```

Reply:
Unfortunately, didn't help